test/inductor/test_compiled_autograd.py

```diff
@@ -0,0 +1,244 @@
+# Owner(s): ["module: inductor"]
+import torch
+from torch import _inductor as inductor
+from torch._dynamo import compiled_autograd
+from torch._dynamo.test_case import run_tests, TestCase
+from torch._dynamo.utils import counters
+from torch.testing._internal.inductor_utils import HAS_CPU
+
+
+def compiler_fn(gm):
+    """Same as torch.compile() but counts number of compiles"""
+
+    def inner_compiler(gm_, example_inputs_):
+        counters["compiled_autograd"]["compiles"] += 1
+        return inductor.compile(gm_, example_inputs_)
+
+    return torch.compile(gm, backend=inner_compiler, fullgraph=True, dynamic=True)
+
+
+# TODO(jansel): hooks as lambdas creates recompiles in dynamo, we should fix that
+
+
+def hook1(grad):
+    return grad * 2
+
+
+def hook2(grads):
+    return (grads[0] + 1,)
+
+
+def hook3(gI, gO):
+    return (torch.sin(gI[0]) + gO[0],)
+
+
+class TestCompiledAutograd(TestCase):
+    def check_output_and_recompiles(self, fn, count=1):
+        with torch.autograd.set_multithreading_enabled(False):
+            torch._dynamo.reset()
+            counters["compiled_autograd"].clear()
+            torch.manual_seed(123)
+            expected = list(fn())
+            torch.manual_seed(123)
+            with compiled_autograd.enable(compiler_fn):
```

Comment:
Does this check if PT was compiled with COMPILED_AUTOGRAD or not?

Reply:
Inductor tests aren't run on Windows.  You think we need a separate flag than just windows?